### PR TITLE
Issue #544 from collection strawman scanLeft and scanRight on ArrayOps

### PR DESCRIPTION
**Pull request by wibisono**
Trying to provide [scanLeft](https://github.com/scala/collection-strawman/issues/544) and also Scan Right

### CODE REVIEW

**Review by julienrf — COMMENTED** [review]
That looks good to me, do you mind also adding `scanRight`?

**Comment by wibisono** [comment]
I will add as soon as I am back from work, I need to practice Scala :) Wondering if you know again why my PR does not trigger builds

File: src/library/scala/collection/ArrayOps.scala [hunk 1]
```diff
@@ -558,6 +558,34 @@ final class ArrayOps[A](val xs: Array[A]) extends AnyVal {
   }
 
 
+  /** Applies a binary operator to a start value and all elements of this array,
+    * going right to left, keeping list of intermediate values.
```
Comment (julienrf) [inline src/library/scala/collection/ArrayOps.scala]: This description is not very clear. What do you think of reusing the existing description? https://github.com/scala/scala/blob/v2.12.5/src/library/scala/collection/GenTraversableLike.scala#L126-L138

**Review by julienrf — COMMENTED** [review]
Apart from the scaladoc, that looks good to me!

File: src/library/scala/collection/ArrayOps.scala (hunk 1 above)
Comment (wibisono) [reply src/library/scala/collection/ArrayOps.scala]: Good idea, it was a copy paste from foldLeft adding a sentence about intermediate values. I think simple example on existing description explains better than x_1, x_2

**Review by lrytz — CHANGES_REQUESTED** [review]
@wibisono some minor git cleanups:
  - Please update the commit message of the first commit so it references https://github.com/scala/collection-strawman/issues/544 (not just `#544`, which then links to an unrelated PR).
  - Also include a description of the change in the commit messages (https://github.com/scala/scala/blob/2.13.x/CONTRIBUTING.md#clean-commits-clean-history)
  - Squash the last commits into one - we try to keep the history "clean" in this repo, we squash changes addressing review comments. Every commit should be self-contained.

Thank you!!

**Comment by lrytz** [comment]
Actually you can squash all the changes into a single commit, as they are all related.

**Comment by wibisono** [comment]
Squashed all commit into one and tried to follow contribution guidelines. Hopefully I learn to commit better next time :)

**Comment by lrytz** [comment]
Thank you very much! Sorry about our stricter-than-usual commit history standards...

**Review by lrytz — APPROVED** [review]

**Comment by wibisono** [comment]
No worries, I think it is better to follow a certain standard. I need to improve my commit message taste :)

**Comment by SethTisue** [comment]
needs rebase

**Comment by wibisono** [comment]
Rebased as suggested

**Review by julienrf — APPROVED** [review]

**Pull request merged**